algolia/search/index.go

```diff
@@ -31,7 +32,7 @@ func newIndex(client *Client, name string) *Index {
 }
 
 func (i *Index) path(format string, a ...interface{}) string {
-	prefix := fmt.Sprintf("/1/indexes/%s", i.name)
+	prefix := fmt.Sprintf("/1/indexes/%s", url.QueryEscape(i.name))
```

Comment:
I am afraid that this may be breaking change for existing customers. 🙁

Reply:
Definitely a breaking change, but a bug fix so there is no way around it.
Currently the behavior is very dangerous as it transforms index names on the fly by truncating starting from the hashtag.

Reply:
Agree with @rayrutjes on this. Changing it is bad for some users on the short-term but not doing anything is bad for all users on the long term. I'm in favor of changing this even if it means introducing a breaking change. I'll check with @robertmogos on the feasability of emailing all Go client users to let them know about the bug and the bug fix. In any case, this will end up in the ChangeLog.

Reply:
Good call @aseure!

Reply:
I'd actually call this a fixing change if that can make everyone feel better :)